### PR TITLE
Created a new function named get_element in GridContainer. This funct…

### PR DESCRIPTION
**Pull request by SASUPERNOVA**
…ion gets the Control at the specified row and column. This works in a similar manner as the operator[] (i.e. indexing starts from 0).

### CODE REVIEW

File: scene/gui/grid_container.cpp [hunk 1]
```diff
@@ -232,6 +233,23 @@ Size2 GridContainer::get_minimum_size() const {
 	return ms;
 }
 
+Control* GridContainer::get_element(int row, int column)
```
Comment (reduz) [inline scene/gui/grid_container.cpp]: would rename to get_child_control_at_cell(int row,int column)

**Review by reduz — CHANGES_REQUESTED** [review]

File: scene/gui/grid_container.cpp [hunk 2]
```diff
@@ -178,6 +178,7 @@ void GridContainer::_bind_methods() {
 
 	ClassDB::bind_method(D_METHOD("set_columns", "columns"), &GridContainer::set_columns);
 	ClassDB::bind_method(D_METHOD("get_columns"), &GridContainer::get_columns);
+	ClassDB::bind_method(D_METHOD("get_element"), &GridContainer::get_element);
```
Comment (Noshyaar) [inline scene/gui/grid_container.cpp]: should be `D_METHOD("get_child_control_at_cell", "row", "column")`

**Comment by Noshyaar** [comment]
There are some styling issues. The changes didn't comply with clang-format. Please use `clang-format -i <filename>` and `git commit --amend`.

**Comment by SASUPERNOVA** [comment]
I had some issues with updating my branch, but I think it should be OK now. Please let me know if the files have not been properly updated.

**Comment by akien-mga** [comment]
There are still merge conflicts + style issues (you changed indentation and line wrappings), see: https://github.com/godotengine/godot/pull/16977/files

Once fixed, the commits should be squashed together to have only one commit changing from the initial state to the new state.

**Comment by akien-mga** [comment]
Thanks, looks good.

**Comment by akien-mga** [comment]
This raises a warning with GCC 8:
```
scene/gui/grid_container.cpp: In member function 'Control* GridContainer::get_child_control_at_cell(int, int)':
scene/gui/grid_container.cpp:256:9: warning: 'c' may be used uninitialized in this function [-Wmaybe-uninitialized]
  return c;
         ^
```

Looking at it again, the whole method seems to be wrong - why use a for loop to iterate over all children when you only want the child with index `grid_index`? You could just use `get_child(grid_index)` directly.

And is it guaranteed that `GridContainer` will always have one child in each cell? If not this method really doesn't make sense. It also doesn't have proper checks that `row` and `column` are within bounds, and can thus return an uninitialized `Control *c` as raised above.

**Comment by akien-mga** [comment]
Reverted in #22822, but it looks like I did misunderstand what the method does and that it's not that trivial, and might be worth readding with a fixed implementation and more comments. See https://github.com/godotengine/godot/pull/22788#discussion_r223178217

**Pull request merged**